filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsResolver.scala

```diff
@@ -76,35 +80,37 @@ object DrsResolver {
   def getContainerRelativePath(drsPath: DrsPath): IO[String] = {
     val pathIO = for {
       drsPathResolver <- getDrsPathResolver(drsPath)
-      tuple <- getGsUriFileNameBondProvider(drsPath.pathAsString, drsPathResolver)
-      (gsUriOption, fileNameOption, _) = tuple
+      localizationData <- getMarthaLocalizationData(drsPath.pathAsString, drsPathResolver)
       /*
       In the DOS/DRS spec file names are safe for file systems but not necessarily the DRS URIs.
       Reuse the regex defined for ContentsObject.name, plus add "/" for directory separators.
       https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/#_contentsobject
        */
       rootPath = DefaultPathBuilder.get(drsPath.pathWithoutScheme.replaceAll("[^/A-Za-z0-9._-]", "_"))
-      fileName <- getFileName(fileNameOption, gsUriOption)
-      fullPath = rootPath.resolve(fileName)
-      fullPathString = fullPath.pathAsString
-    } yield fullPathString
+      containerRelativePath <- buildContainerRelativePath(localizationData, rootPath)
+    } yield containerRelativePath.pathAsString
 
     pathIO.handleErrorWith(resolveError(drsPath.pathAsString))
   }
 
-  /**
-    * Return the file name returned from the martha response or get it from the gsUri
-    */
-  private def getFileName(fileName: Option[String], gsUri: Option[String]): IO[String] = {
-    fileName match {
-      case Some(actualFileName) => IO.pure(actualFileName)
-      case None =>
-        //Currently, Martha only supports resolving DRS paths to GCS paths
+  // Return the container relative path built from the Martha-specified localization path, file name, or gs URI.
+  private def buildContainerRelativePath(localizationData: MarthaLocalizationData, rootPath: Path): IO[Path] = {
```

Comment:
`rootPath` is the directory that the DRS file was always localized to before these changes, build to resemble the DRS URI, right? I found the distinction between `rootPath` and container root confusing here. Is it like:
 * If we get a localizationPath, copy to `${containerRoot}/${localizationPath}`
 * If we do NOT get a localizationPath, copy to`${containerRoot}/${mungedDRSPathString}/${filename}`

We might want to call `rootPath` something different, now that it's not necessarily the root for all files. `defaultLocalDir`? . I don't want to block this PR on that, though.

Reply:
Yes I think you're right, I'll look at cleaning this up.